src/snowflake/cli/api/console/console.py

```diff
@@ -0,0 +1,43 @@
+from __future__ import annotations
+
+from snowflake.cli.api.cli_global_context import cli_context
+from snowflake.cli.api.console.abc import AbstractConsole
+from snowflake.cli.api.console.enum import Output
+
+
+class CliConsole(AbstractConsole):
```

Comment:
the phase/step API is interesting. It looks like top-level steps are allowed, but as soon as a phase is used, then top level steps are no longer allowed. So something like:
- step 1
- long phase
  - step a
  - step b
- step 2

cannot be output, is that right?

Also, it looks like there is no way to indicate a success / error for a step. Is that something you've considered? It's rather common to use something like a spinner that resolves to some kind of OK/Error indicator when the task is complete, which visually a little more intuitive for a user. If you're doing something like a parallel upload to a stage, having the ability to indicate progress for each file might come in handy, for instance.

Reply:
Any reason why we ended up with only phase and step? It seems a bit too restrictive to only allow one/2 indentations. Is there anything specific to being in a phase or in a step except the indentation? and if not, can't we just accept indentation value as a parameter?

Reply:
The goal of separate `phase` and `step` was to display extra information while commands are executed:
- `phase` to be used for main events 
- `step` to be used when some detailed information needed to be displayed for main event - indentation.

more like `heading 1` and  `hading 2` format.

The case @sfc-gh-bdufour introduced may suggest people will use this methods interchangeably and keeping track in mind which method is intended to produce what kind of output is an unnecessary complexity.

I think a solution with less overhead for dev would be a simple `print` like method with a param to indicate whether output should be indented or not.

consider this two examples:

```python
import cli_console as cc

cc.phase("Task A")
cc.phase("Task B")
cc.step("Task B: part 1")
cc.step("Task B: part 2")
cc.exit_phase()
cc.step("Some C")
```

vs
```python
import cli_console a cc

cc.print("Task A")
cc.print("Task B")
cc.print("Task B: part 1", indent=True)   # or cc.print_indented("Task B: part 1")
cc.print("Task B: part 2", indent=True)   # or cc.print_indented("Task B: part 2")
cc.print("Some C")
```

WDYT?

Reply:
The main reason to have phase/step instead of print is consistency. With growing number of developers working on snowcli we want to assure that if we show progress, then we show it in the same way across whole cli. 

Yes, `print` is a quick win but that's not our goal. We don't want to have a script with bunch of commands. We want to have nice cli tool.

Reply:
> Any reason why we ended up with only phase and step? It seems a bit too restrictive to only allow one/2 indentations. Is there anything specific to being in a phase or in a step except the indentation? and if not, can't we just accept indentation value as a parameter?

The goal is to establish an interface for unified output of intermediate progress. 
We want to:
- have a control over the way the output is presented to users,
- be able to easily detect cases where output could be corrupted or create side effects.

The idea behind `phase` and `step` is to was to create predefined formatted output.
`phase` is ment to be used for displaying unindented and `step` for indented output.

Indentation tracking feature brings confusion and it will be removed for now.
There is always a possibility of futher development of console output that will result in more features for displaying output.

For now we will provide:
- a `phase` method for unindented output with predefined color
- a `step` method for 2 spaces indented output with predefined color
- a generic `error` method for error messages with no indentation and predefined color

Reply:
Thanks for this. I'm not sure I follow the mental model behind the mapping then, other than phase = not indented message, and step = indented message. There seems to be no real semantic difference between them in the sample code you wrote. That worries me a little. Would you be open to whiteboarding some ideas next week since we'll be physically in the same location?

Reply:
> Thanks for this. I'm not sure I follow the mental model behind the mapping then, other than phase = not indented message, and step = indented message. There seems to be no real semantic difference between them in the sample code you wrote. That worries me a little. Would you be open to whiteboarding some ideas next week since we'll be physically in the same location?

I'll be happy to discuss the ideas,
I think we could schedule something remotely this week.